cpp/src/io/parquet/page_data.cu

```diff
@@ -1329,44 +1341,43 @@ static __device__ void gpuUpdateValidityOffsetsAndRowIndices(int32_t target_inpu
       PageNestingInfo *pni = &s->page.nesting[s_idx];
 
       // if we are within the range of nesting levels we should be adding value indices for
-      int in_nesting_bounds =
+      int const in_nesting_bounds =
         ((s_idx >= start_depth && s_idx <= end_depth) && in_row_bounds) ? 1 : 0;
 
       // everything up to the max_def_level is a non-null value
-      uint32_t is_valid = 0;
-      if (d >= pni->max_def_level && in_nesting_bounds) { is_valid = 1; }
+      uint32_t const is_valid = d >= pni->max_def_level && in_nesting_bounds ? 1 : 0;
 
       // compute warp and thread valid counts
-      uint32_t warp_valid_mask;
-      // for flat schemas, a simple ballot_sync gives us the correct count and bit positions because
-      // every value in the input matches to a value in the output
-      if (max_depth == 0) {
-        warp_valid_mask = ballot(is_valid);
-      }
-      // for nested schemas, it's more complicated.  This warp will visit 32 incoming values,
-      // however not all of them will necessarily represent a value at this nesting level. so the
-      // validity bit for thread t might actually represent output value t-6. the correct position
-      // for thread t's bit is cur_value_count. for cuda 11 we could use __reduce_or_sync(), but
-      // until then we have to do a warp reduce.
-      else {
-        warp_valid_mask = WarpReduceOr32(is_valid << thread_value_count);
-      }
+      uint32_t const warp_valid_mask =
+        // for flat schemas, a simple ballot_sync gives us the correct count and bit positions
+        // because every value in the input matches to a value in the output
+        max_depth == 1
```

Comment:
Was the inversion of the if here intended?

Reply:
Sort of.  It should have been this way all along.  The computation ends up the same, it's just that doing it with the warp reduce is more expensive. Whoops.

Reply:
I'm having difficulty understanding this. Isn't `max_depth` for flat schema 0?

Reply:
No. max_depth is semantically similar to "size".  A simple int column has a depth of 1.  A list<int> is 2, etc.  Depth doesn't mean "depth past the first level" it just means "height of the hierarchy".